### DISCUSSION TITLE
Restarting with checkpoints : huge generated data, procs number

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#28933] [Show and tell]
Hello,

In the "Small Strain" framework, we typically restart jobs by utilizing the [UserObjects] block or by prescribing initial conditions (IC) on the desired variables. However, this approach has not been feasible in the "Finite Strain" framework, and we are unclear why it's impossible to apply an IC condition in this context.

To address this, we have switched to using checkpoints for restarting jobs. This method, however, results in the creation of numerous large subfolders  and several gigabytes of data. Additionally, a limitation we've encountered is the inability to change the number of processors (procs) between the initial job and the restart—running the first job with X procs means we must use the same number for subsequent jobs.

We have reviewed the "Restart and Recover" documentation but have not found an effective solution to these issues. Could you please provide guidance on how to reduce the volume of generated data and advise on a strategy to run jobs with varying numbers of procs?

Thank you for your assistance.

**Comment by selarem** [comment]
Could anyone help in explaining to us why it is not possible to apply IC in  solid mechanics within the   "finite strain" assumption ?

**Reply by GiudGiud** [reply]
ICs are for variables. You need to be restarting stresses iirc, which are material properties

Checkpoints are the correct way to restart these simulations.

**Reply by sapitts** [reply]
You could alternatively try applying an eigenstrain at the start of the simulation, although depending on your strain field "your mileage may vary". @dewenyushu may have some additional insights, if the initial stress method @GiudGiud suggested will not work for your case

**Reply by selarem** [reply]
While we have successfully resolved our issue using checkpoints, we still lack clarity on why initial conditions (ICs) fail to function as expected. For any given prescribed displacement, the solver is supposed to iterate until it converges. This is not happening in our case. It only works in "small strain".

**Reply by GiudGiud** [reply]
>  Could you please provide guidance on how to reduce the volume of generated data and advise on a strategy to run jobs with varying numbers of procs?

You can set the number of checkpoint files to 1 instead of 2 to reduce the amount of data by a factor of 2.
Jobs with varying number of procs wont be supported with checkpoints. 

> we still lack clarity on why initial conditions (ICs) fail to function as expected

The finite strain approach is relying on this object right?
https://mooseframework.inl.gov/source/materials/ADComputeFiniteStrain.html

If so that's a stateful material property, which ICs cannot restart.
